src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/policy/model/Criterion.kt

```diff
@@ -1,15 +1,26 @@
 package uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.service.policy.model
 
+fun interface EvaluationStrategy {
+  fun isMet(answers: Map<String, Boolean>): Boolean
+
+  companion object {
+    val MET_IF_ALL_ARE_FALSE = EvaluationStrategy { answers -> answers.values.all { !it } }
+    val MET_IF_ANY_ARE_FALSE = EvaluationStrategy { answers -> answers.values.any { !it } }
+  }
+}
+
 data class Criterion(
   val code: String,
   val name: String,
   val questions: List<Question> = emptyList(),
-) {
+  val evaluationStrategy: EvaluationStrategy = EvaluationStrategy.MET_IF_ALL_ARE_FALSE,
+) : EvaluationStrategy by evaluationStrategy {
```

Comment:
think this is the first time I've used kotlin delegation

Reply:
rather than having to say that Criterion implements evaluation strategy and then 

`fun isMet(questions: Map<string, boolean>) = evaluationStrategy.isMet(questions)`

Reply:
yeah I don't think I've ever seen it used outside of a book. Sure it's useful once you get used to it